core/src/main/java/org/elasticsearch/cluster/metadata/SnapshotId.java

```diff
@@ -19,37 +19,78 @@
 
 package org.elasticsearch.cluster.metadata;
 
+import org.elasticsearch.common.UUIDs;
 import org.elasticsearch.common.io.stream.StreamInput;
 import org.elasticsearch.common.io.stream.StreamOutput;
-import org.elasticsearch.common.io.stream.Streamable;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.snapshots.Snapshot;
 
 import java.io.IOException;
+import java.util.Objects;
 
 /**
- * Snapshot ID - repository name + snapshot name
+ * SnapshotId - snapshot (repository name + snapshot name) + snapshot UUID
  */
-public class SnapshotId implements Streamable {
+public class SnapshotId implements Writeable {
```

Comment:
Unfortunately I can't.  I tried it earlier, but mockito uses this class in some tests and mockito can't work with `final` classes